scr/main/java/bittercode/util/StoreUtil.java

```diff
@@ -0,0 +1,86 @@
+    }
+
+    /**
+     * Modify the active tab in the page menu bar
+     */
+    public static void setActiveTab(PrintWriter pw, String activeTab) {
+
+        pw.println("<script>document.getElementById(activeTab).classList.remove(\"active\");activeTab=" + activeTab
+                + "</script>");
+        pw.println("<script>document.getElementById('" + activeTab + "').classList.add(\"active\");</script>");
+
+    }
+
+    /**
+     * Add/Remove/Update Item in the cart using the session
+     */
+    public static void updateCartItems(HttpServletRequest req) {
+        String selectedBookId = req.getParameter("selectedBookId");
+        HttpSession session = req.getSession();
+        if (selectedBookId != null) { // add item to the cart
+
+            // Items will contain comma separated bookIds that needs to be added in the cart
+            String items = (String) session.getAttribute("items");
+            if (req.getParameter("addToCart") != null) { // add to cart
+                if (items == null || items.length() == 0)
+                    items = selectedBookId;
+                else if (!items.contains(selectedBookId))
+                    items = items + "," + selectedBookId; // if items already contains bookId, don't add it
+
+                // set the items in the session to be used later
+                session.setAttribute("items", item);
+
+                /*
+                 * Quantity of each item in the cart will be stored in the session as:
+                 * Prefixed with qty_ following its bookId
+                 * For example 2 no. of book with id 'myBook' in the cart will be
+                 * added to the session as qty_myBook=2
+                 */
+                int itemQty = 0;
+                if (session.getAttribute("qty_" + selectedBookId) != null)
+                    itemQty = (int) session.getAttribute("qty_" + selectedBookId);
+                itemQty += 1;
+                session.setAttribute("qty_" + selectedBookId, itemQty);
+            } else { // remove from the cart
+                int itemQty = 0;
+if (session.getAttribute("qty_" + selectedBookId) != null)
+                    itemQty = (int) session.getAttribute("qty_" + selectedBookId);
+                if (itemQty == 1) {
+                    itemQty--;
+                    session.setAttribute("qty_" + selectedBookId, itemQty);
+                } else {
+                    session.removeAttribute("qty_" + selectedBookId);
+                    items = items.replace(selectedBookId + ",", "");
+                    items = items.replace("," + selectedBookId, "");
+                    items = items.replace(selectedBookId, "");
+                    session.setAttribute("items", items);
+                }
+                else  {
+                    session.removeAttribute("qty_" + selectedBookId);
+                    session.setAttribute("items", items.replace(selectedBookId, ""));
+                }
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider consolidating cart removal logic</b></div>
<div id="fix">

The logic for removing items from cart has redundant code paths and incorrect control flow. The `else` block after line 77 is unreachable and duplicates functionality. Consider consolidating the item removal logic.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">

```
 @@ -68,15 +68,10 @@
                  if (itemQty == 1) {
                      itemQty--;
                      session.setAttribute("qty_" + selectedBookId, itemQty);
 -                } else {
 +                } else if (itemQty > 1) {
                      session.removeAttribute("qty_" + selectedBookId);
                      items = items.replace(selectedBookId + ",", "");
                      items = items.replace("," + selectedBookId, "");
                      items = items.replace(selectedBookId, "");
                      session.setAttribute("items", items);
 -                }
 -                else  {
 -                    session.removeAttribute("qty_" + selectedBookId);
 -                    session.setAttribute("items", items.replace(selectedBookId, ""));
                  }
```


```suggestion
                if (itemQty == 1) {
                    itemQty--;
                    session.setAttribute("qty_" + selectedBookId, itemQty);
                } else if (itemQty > 1) {
                    session.removeAttribute("qty_" + selectedBookId);
                    items = items.replace(selectedBookId + ",", "");
                    items = items.replace("," + selectedBookId, "");
                    items = items.replace(selectedBookId, "");
                    session.setAttribute("items", items);
                }
```



</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/7#issuecomment-2527741727>#f9242c</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged